modules/secrets-store-azure.adoc

```diff
@@ -0,0 +1,313 @@
+
+.Prerequisites
+
+* Your cluster is installed on Azure.
+* You have installed the {secrets-store-operator}. See _Installing the {secrets-store-driver}_ for instructions.
+* You have configured Azure Key Vault to store the required secrets.
+* You have installed the Azure CLI (`az`).
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Install the Azure Key Vault provider:
+
+.. Create a YAML file with the following configuration for the provider resources:
++
+.Example `azure-provider.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: csi-secrets-store-provider-azure
+  namespace: openshift-cluster-csi-drivers
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: csi-secrets-store-provider-azure-cluster-role
+rules:
+- apiGroups: [""]
+  resources: ["serviceaccounts/token"]
+  verbs: ["create"]
+- apiGroups: [""]
+  resources: ["serviceaccounts"]
+  verbs: ["get"]
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get"]
+- apiGroups: [""]
+  resources: ["nodes"]
+  verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: csi-secrets-store-provider-azure-cluster-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: csi-secrets-store-provider-azure-cluster-role
+subjects:
+- kind: ServiceAccount
+  name: csi-secrets-store-provider-azure
+  namespace: openshift-cluster-csi-drivers
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  namespace: openshift-cluster-csi-drivers
+  name: csi-secrets-store-provider-azure
+  labels:
+    app: csi-secrets-store-provider-azure
+spec:
+  updateStrategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: csi-secrets-store-provider-azure
+  template:
+    metadata:
+      labels:
+        app: csi-secrets-store-provider-azure
+    spec:
+      serviceAccountName: csi-secrets-store-provider-azure
+      hostNetwork: true
+      containers:
+        - name: provider-azure-installer
+          image: mcr.microsoft.com/oss/azure/secrets-store/provider-azure:v1.4.1
+          imagePullPolicy: IfNotPresent
+          args:
+            - --endpoint=unix:///provider/azure.sock
+            - --construct-pem-chain=true
+            - --healthz-port=8989
+            - --healthz-path=/healthz
+            - --healthz-timeout=5s
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: 8989
+            failureThreshold: 3
+            initialDelaySeconds: 5
+            timeoutSeconds: 10
+            periodSeconds: 30
+          resources:
+            requests:
+              cpu: 50m
+              memory: 100Mi
+            limits:
+              cpu: 50m
+              memory: 100Mi
+          securityContext:
+            privileged: true
```

Comment:
Currently this does not work on ARO clusters if we secretproviderclass with deployment kind object. 
```suggestion
            allowPrivilegeEscalation: false
            readOnlyRootFilesystem: true
            runAsUser: 0
            capabilities:
              drop:
              - ALL
```
Discussion: https://redhat-internal.slack.com/archives/C05CXR1PVRD/p1694708747327749?thread_ts=1694466510.194569&cid=C05CXR1PVRD

Reply:
I updated to remove `privileged: true`, and added these lines instead.

Will this allow the deployment to work, or are you still saying that we need to use a pod in this procedure for Azure?

Reply:
As per https://github.com/openshift/openshift-docs/pull/62385#discussion_r1337240518 , if deployment and/or other replica controllers (ds, rs, job et al) do not work with secretproviderclass, a bug should be filed.

Reply:
As per https://github.com/openshift/openshift-docs/pull/62385#discussion_r1337240518 , if deployment and/or other replica controllers (ds, rs, job et al) do not work with secretproviderclass, a bug should be filed, CC @duanwei33 .